docs: add to documentation for Kubernetes deployments, devcontainer improvements (#498)
ai-dynamo/dynamo: deploy/Kubernetes/pipeline/README.md

- make sure dynamo cli is installed
- make sure you have a docker image registry to which you can push and pull from k8s cluster
- set the imagePullSecrets in the values.yaml file

> - navigate to the pipeline deployment directory by running:

Reviewer: @statiraju - should these go into the parent `docs/` folder so that they render?

Reviewer 2: @ishandhanani i am fine with @hhzhang16 merging this. I can move this along with other READMEs and fix on my PR

  ```bash
  cd deploy/Kubernetes/pipeline
  ```

> - build and set up the DYNAMO_IMAGE as described in the [main README](../../README.md#building-the-dynamo_image-base-image)

Reviewer 3: ```suggestion
- build and push the DYNAMO_IMAGE as described in the [main README](../../README.md#building-the-dynamo_image-base-image) to an image registry
```

- make sure the `nats` and `etcd` dependencies are installed (under the `dependencies` subdirectory). For more details, see [Installing Required Dependencies](../../../docs/guides/dynamo_deploy.md#installing-required-dependencies)

### Install the Helm chart

```bash

> export DYNAMO_IMAGE=<dynamo_docker_image_name>

Reviewer 3: This needs to be set to the image on your remote registry. (Doesn't work if it's a local image)

> ./deploy.sh <docker_registry> <k8s_namespace> <path_to_dynamo_directory> <dynamo_identifier> [<dynamo_config_file>]

Reviewer 3: Should we add a section about how to set the imagepullsecret within the helm chart? Assuming they pushed to a registry, the resources would need to reference or create an imagepullsecret to pull this base image, correct?

Author: I mention it here: https://github.com/ai-dynamo/dynamo/tree/main/deploy/dynamo/helm/README.md (it was originally in this MR but got pushed in with yours haha), I can mention it here too if you want!

Reviewer 3: What you've linked is how to setup image pull secret for Dynamo cloud. I believe this deploy script refers to directly deploying with helm. I think what we still need to call out setting an imagepullsecret for the latter case.

Reviewer 3: What is the imagepullsecret the helm chart uses by default?

Author: I think the docs would be similar so I'll just write something similar -- my vision was to have the user use `docker-imagepullsecret`
